web/src/dialog_widget.ts

```diff
@@ -102,6 +102,26 @@ export function close(on_hidden_callback?: () => void): void {
     modals.close("dialog_widget_modal", {on_hidden: on_hidden_callback});
 }
 
+export function get_current_values($inputs: JQuery): Record<string, unknown> {
+    const current_values: Record<string, unknown> = {};
+    $inputs.each(function () {
+        const property_name = $(this).attr("name")!;
+        if (property_name) {
+            if ($(this).is("input[type='file']") && $(this).prop("files")?.length) {
+                // If the input is a file input and a file has been selected, set value to file object
+                current_values[property_name] = $(this).prop("files")[0];
+            } else if (property_name === "edit_bot_owner") {
+                current_values[property_name] = $(this).find(".dropdown_widget_value").text();
+            } else if (property_name === "Mentor") {
```

Comment:
This is not correct. Different organizations can have different names for a user type field. We should do something like what we do for file type fields here.

Also, this change should be done in the commit which disables and enables the button in user profile modal. This commit should only have changes to extract the function.

Reply:
Fixed it.